prime-router/src/main/kotlin/fhirengine/utils/HL7MessageHelpers.kt

```diff
@@ -32,6 +32,22 @@ object HL7MessageHelpers : Logging {
     fun batchMessages(hl7RawMsgs: List<String>, receiver: Receiver): String {
         check(receiver.translation is Hl7Configuration)
         val useBatchHeaders = receiver.translation.useBatchHeaders
+        val replaceValueAwithB = receiver.translation.replaceValueAwithB ?: emptyMap()
+        var hl7FileHeaderEncodingCharInReplace: String? = hl7BatchHeaderEncodingChar
```

Comment:
Rather than looking at a new translation setting, you could alternatively look at `MSH.2` so that we can't produce batches where those values fall out of sync.

Reply:
Thank you.

Reply:
I also agree with Michael. We should be able to use the same logic for geting MSH-5 like this:
`val hl7EncodingCharacters =  firstMessage?.get("MSH-2") ?: hl7BatchHeaderEncodingChar `

Reply:
@victor-chaparro , @mkalish Done.  Thank you very much.